### PR TITLE
Removing old comments

### PR DESCRIPTION
**Pull request by BrianBohe**
Removing two TODO comments.

1) The problem of lifetime is solved since [a variable is reported each time of its lifetime in different position (register or stack)](https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/jit/ee_il_dll.cpp#L638-L661)

2) XMM [are defined](https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/jit/target.h#L289-L305) and [used in some places](https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/jit/codegen.h#L905-L995)

### CODE REVIEW

**Comment by BrianBohe** [comment]
@CarolEidt @BruceForstall

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -10383,10 +10383,6 @@ void CodeGen::genSetScopeInfo()
 
         Compiler::siVarLoc varLoc;
 
-        // TODO-Review: This only works for always-enregistered variables. With LSRA, a variable might be in a register
-        // for part of its lifetime, or in different registers for different parts of its lifetime.
-        // This should only matter for non-debug code, where we do variable enregistration.
-        // We should store the ranges of variable enregistration in the scope table.
```
Comment (CarolEidt) [inline src/jit/codegencommon.cpp]: I'm not sure this comment is obsolete. As far as I can tell, we still only report variable locations at scope boundaries, and any register candidate will be reported as whatever register it happens to occupy at the end of the method.

Comment (BrianBohe) [reply src/jit/codegencommon.cpp]: Yes, you are right. I revert that commit. Thanks!

File: src/jit/codegencommon.cpp [hunk 2]
```diff
@@ -10438,8 +10438,6 @@ void CodeGen::genSetScopeInfo()
 
                 case TYP_FLOAT:
                 case TYP_DOUBLE:
-                    // TODO-AMD64-Bug: ndp\clr\src\inc\corinfo.h has a definition of RegNum that only goes up to R15,
```
[the remark this answers is not in the document]
Comment (BrianBohe) [reply src/jit/codegencommon.cpp]: Do you know what are be the changes needed to report those registers?

**Pull request closed without merging**